forte/data/ontology/top.py

```diff
@@ -1261,6 +1267,15 @@ def __setstate__(self, state):
         self.__dict__.update(state)
         self._modality = getattr(Modality, state["_modality"])
 
+        # During de-serialization, convert the list back to numpy array.
+        if "_embedding" in state:
+            state["_embedding"] = np.array(state["_embedding"])
+        else:
+            state["_embedding"] = np.empty(0)
+
+        if "_cache" in state and isinstance(state["_cache"], list):
```

Comment:
This checking might not be stable. Can we assume that `_cache` is a numpy array when it was stored as a list in serialization format?

Reply:
I think we can. Because if `_cache` is not stored as a string (which is in the case of text), it is stored an an `ndarray`. It can be seen by the definition of `set_cache` [here](https://github.com/asyml/forte/blob/8a61272fe2dd7d3f266f4d66a2f0d3e8f6fc3389/forte/data/ontology/top.py#L1226)

Reply:
Cool. Then let's just add the assumption here in the comments since it might not be true in future when we have a more flexible Payload that can stores any type of data.